packages/synapse-react-client/src/components/WikiMarkdownEditor/WikiMarkdownEditor.tsx

```diff
@@ -0,0 +1,256 @@
+import { Alert, Box, Button, TextField } from '@mui/material'
+import { ObjectType, WikiPageKey } from '@sage-bionetworks/synapse-types'
+import { noop } from 'lodash-es'
+import React, { useEffect, useMemo, useState } from 'react'
+import {
+  CreateWikiPageInput,
+  UpdateWikiPageInput,
+  useCreateWikiPage,
+  useGetRootWikiPageKey,
+  useGetWikiPage,
+  useUpdateWikiPage,
+} from '../../synapse-queries'
+import ConfirmationDialog from '../ConfirmationDialog'
+import { DialogBase } from '../DialogBase'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import { MarkdownEditor } from '../Markdown'
+
+export const UNSAVED_CHANGES = 'Unsaved Changes'
+export const NAVIGATE_AWAY_CONFIRMATION_MESSAGE =
+  'Any unsaved changes may be lost. Are you sure that you would like to navigate away from this editor?'
+export const ERROR_LOADING_WIKI_FAILED = 'Failed to load the wiki page: '
+export const ERROR_SAVING_WIKI =
+  'Could not save your changes. It is recommended that you copy your version of the wiki text so that it is not lost. '
+
+export type WikiMarkdownEditorProps = {
+  ownerObjectType: ObjectType
+  ownerObjectId: string
+  // if wikiPageId is undefined, will get (or create) the root WikiPage for this ownerObject
+  // otherwise, will get the WikiPage with the specified wikiPageId
+  wikiPageId?: string
+  onCancel?: () => void
+  onSave?: () => void
+  // TODO: SWC-6774 - expose delete prop
+  // showDeleteButton: boolean
+}
+
+export const WikiMarkdownEditor: React.FunctionComponent<
+  WikiMarkdownEditorProps
+> = (props: WikiMarkdownEditorProps) => {
+  // TODO: SWC-6774 - get showDeleteButton from props
+  const showDeleteButton = false
+  const {
+    ownerObjectType,
+    ownerObjectId,
+    wikiPageId: initialWikiPageId,
+    onCancel = noop,
+    onSave = noop,
+  } = props
+
+  const [wikiPageId, setWikiPageId] = useState<string | undefined>(
+    initialWikiPageId,
+  )
+  const [open, setOpen] = useState<boolean>(true)
+  const [title, setTitle] = useState<string>('')
+  const [markdown, setMarkdown] = useState<string>('')
+  const [showConfirmCancelDialog, setShowConfirmDialog] =
+    useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
```

Comment:
Is it possible to simplify the logic in your effects by deriving the `error` in one function/hook, instead of setting it via callbacks + effects?

i.e. 
```ts

// pull the error out of the mutation hooks instead of using `onError`
const { mutate: createWikiPage, error: createError } = useCreateWikiPage({
	// ...
})

const error = useMemo(() => {
	if (errorLoadingRootWikiPageKey) {
 		return ERROR_LOADING_WIKI_FAILED // + whatever
	} else if (createError) {
		return createError.reason // or whatever
	} else if (updateError) {
		 //...
	}
	return null;
})

```

Your use memo logic might need to be slightly more complicated than this example, but it would make it easier to trace where an error comes from